internal/core/app_status_report.go

```diff
@@ -74,7 +74,54 @@ func (a *App) DeploymentStatusReport(
 	}
 	defer c.Close()
 
-	return a.statusReport(ctx, "deploy_statusreport", c, deployTarget, lastResp)
+	// Create the initial report. Assuming no error, we'll query ListInstances
+	// to find how many active Entrypoint connections are registered for this
+	// deployment. If there is a non-nil error, just return immediately.
+	report, rErr := a.statusReport(ctx, "deploy_statusreport", c, deployTarget, lastResp)
+	if rErr != nil {
+		return report, rErr
+	}
+
+	// Currently the statusReport() call above will only return a nil report if
+	// the error is NOT nil, but it is not an exported method so we guard here
+	// in case that changes.
+	if report != nil {
+		// check if we have any connected instances
+		resp, err := a.client.ListInstances(ctx, &pb.ListInstancesRequest{
```

Comment:
We had a question here for the wider team - It would be nice if Status Reports had some kind of pre-Upsert hook for adding information like active entrypoint connections to a report that a plugin typically wouldn't have access to at the time the Status Report is generated. @catsby and I figured _for now_ since it's just one extra piece of data, it's probably OK to do a double Upsert and append this data later after the plugin has already upserted its report.

In the future though, it would be nice if we had something a bit more generic in place with our system to add data like this. But if someone has a good idea of how to easily achieve that, we're all ears!

Finally - maybe a double Upsert isn't a huge deal here if the cost is low to modify an existing report in the db with a little bit of extra information here.